packages/realm/src/app-services/SyncSession.ts

```diff
@@ -53,13 +53,29 @@ export enum ProgressMode {
   ForCurrentlyOutstandingWork = "forCurrentlyOutstandingWork",
 }
 
-export type ProgressNotificationCallback =
+/** @deprecated */
+export type PartitionBasedSyncProgressNotificationCallback =
   /**
+   * This notification callback only supports Partition Based Sync.
    * @param transferred - The current number of bytes already transferred
    * @param transferable - The total number of transferable bytes (i.e. the number of bytes already transferred plus the number of bytes pending transfer)
+   * @deprecated - This notification callback will be removed in next major release. Use {@link DynamicProgressNotificationCallback} instead.
+   * @since 1.12.0
    */
   (transferred: number, transferable: number) => void;
 
+export type DynamicProgressNotificationCallback =
```

Comment:
In this context `Dynamic` may be ambiguous. It may seem implicit since the progress itself is dynamic, whic PBS also is. But I understand you're referring to PBS-only vs Flx-or-PBS. At another point I saw you used `isEstimateProgress..`. What do you think of that name instead (`EstimateProgressNotificationCallback`)?